lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -160,6 +160,38 @@ class BamDataResult(DataResult):
     max_high: int
 
 
+class DeleteDatasetBatchPayload(BaseModel):
+    ids: List[EncodedDatabaseIdField] = Field(
+        description="The list of datasets IDs to be deleted/purged.",
```

Comment:
I think that as it is you can only delete `HistoryDatasetAssociation` items. I would add the `src` parameter here to control which model you can delete.

Reply:
Hopefully this is addressed in https://github.com/galaxyproject/galaxy/pull/13113/commits/3502fa15486a2dfbc6bda06864b2173fc9afadff

Reply:
I guess the question is would we want to batch-delete mixed content ? If yes, I'd accept something like `[{'src': 'hda'}, 'id': 1], {'src': 'ldda'}, 'id': 2]` etc. It is less efficient in terms of having to repeat `src`, but on the other hand it does feel consistent with regards to other endpoints ... not sure.

Reply:
Ohh I see, then it's better to make it like the other endpoints. Otherwise, it may be too tailored for this particular use case and less flexible.

Reply:
(Just to clarify after reading again what I wrote: I should have written "other, single value payloads in other endpoints", not "other endpoints", we don't have many other batch endpoints)